mfr/extensions/pdf/export.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+import imghdr
+from PIL import Image
+
+from reportlab.pdfgen import canvas
+
+from mfr.core import extension
+from mfr.extensions.pdf import exceptions
+
+
+class PdfExporter(extension.BaseExporter):
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.metrics.add('pil_version', Image.VERSION)
+
+    def tiff_to_pdf(self, tiff_img, max_size):
+        max_pages = 40
+        width, height = tiff_img.size
+        c = canvas.Canvas(self.output_file_path)
+        c.setPageSize((max_size[0], max_size[1]))
+
+        page = 0
+
+        # This seems to be the only way to write this loop at the moment
+        while True:
+            try:
+                tiff_img.seek(page)
+            except EOFError:
+                break
+
+            # Center the image and draw it in the canvas
+            c.drawInlineImage(tiff_img, (max_size[0] - width) // 2, (max_size[1] - height) // 2, anchor='c')
+            # move to the next page
+            c.showPage()
+
+            if max_pages and page > max_pages:
+                break
+            page += 1
+
+        c.save()
+
+    def export(self):
+        parts = self.format.split('.')
+        type = parts[-1].lower()
+        max_size = [int(x) for x in parts[0].split('x')] if len(parts) == 2 else None
+
+        self.metrics.merge({
+            'type': type,
+            'max_size_w': max_size[0],
+            'max_size_h': max_size[1],
+        })
+        try:
+            image = Image.open(self.source_file_path)
+            if max_size:
+                # resize the image to the w/h maximum specified
+                ratio = min(max_size[0] / image.size[0], max_size[1] / image.size[1])
+                self.metrics.add('ratio', ratio)
+                if ratio < 1:
+                    image = image.resize((round(image.size[0] * ratio), round(image.size[1] * ratio)), Image.ANTIALIAS)
+
+            self.tiff_to_pdf(image, max_size)
+            image.close()
+
+        except (UnicodeDecodeError, IOError) as err:
```

Comment:
The code and exception message looks similar. Why not combining the two `except` in one? (A question not a suggestion.)

Reply:
The `ValueError` case is a bug with our current version of pillow having trouble opening certain formats of `.tiff`. Once pillow is updated to 4.3 or higher, this last error section can go away. (tested now and this is indeed the case. `broken.tiff` will open on pillow 4.3, but not 2.8.3 that we currently have. This is why the errors were separate. The tiff given was valid, just pillow messing up.